File: 2442-count-number-of-distinct-integers-after-reverse-operations/2442-count-number-of-distinct-integers-after-reverse-operations.java
class Solution {
    
    
    public int countDistinctIntegers(int[] nums) {
        //list 2개에 reverse한거 합치고, 셋으로 바꾼뒤 size구하기
        Set<Integer> set = new HashSet<>();
        
        for(int n: nums){
            set.add(n);
            set.add(reverse(n));
        }
        
        return set.size();
    }

    int reverse(int n){
        int reverseVal = 0;
        
        while(n != 0){
            int rem = n % 10;
            reverseVal = reverseVal* 10 + rem;
            n /= 10;
        }
        
        return reverseVal;
    }
}

File: 2442-count-number-of-distinct-integers-after-reverse-operations/2442-count-number-of-distinct-integers-after-reverse-operations.kt
class Solution {
fun countDistinctIntegers(nums: IntArray): Int {
    nums.sort()
    var count = HashMap<Int, Int>()
    
    nums.forEach { num -> 
        count.put(num, count.getOrDefault(num, 0) + 1)
    }
    nums.forEach { num -> 
        val reverse = num.toString().reversed().toInt()
        count.put(reverse, count.getOrDefault(reverse, 0) + 1)
    }
    return count.size
}
}